jetty-slf4j-impl/src/main/java/org/eclipse/jetty/logging/JettyLogger.java

```diff
@@ -0,0 +1,663 @@
+        {
+            getAppender().emit(asEvent(Level.WARN, format, arg));
+        }
+    }
+
+    @Override
+    public void warn(String format, Object... arguments)
+    {
+        if (isWarnEnabled())
+        {
+            getAppender().emit(asEvent(Level.WARN, format, arguments));
+        }
+    }
+
+    @Override
+    public void warn(String format, Object arg1, Object arg2)
+    {
+        if (isWarnEnabled())
+        {
+            getAppender().emit(asEvent(Level.WARN, format, arg1, arg2));
+        }
+    }
+
+    @Override
+    public void warn(String msg, Throwable throwable)
+    {
+        if (isWarnEnabled())
+        {
+            getAppender().emit(asEvent(Level.WARN, msg, throwable));
+        }
+    }
+
+    @Override
+    public boolean isWarnEnabled(Marker marker)
+    {
+        return isWarnEnabled();
+    }
+
+    @Override
+    public void warn(Marker marker, String msg)
+    {
+        warn(msg);
+    }
+
+    @Override
+    public void warn(Marker marker, String format, Object arg)
+    {
+        warn(format, arg);
+    }
+
+    @Override
+    public void warn(Marker marker, String format, Object arg1, Object arg2)
+    {
+        warn(format, arg1, arg2);
+    }
+
+    @Override
+    public void warn(Marker marker, String format, Object... arguments)
+    {
+        warn(format, arguments);
+    }
+
+    @Override
+    public void warn(Marker marker, String msg, Throwable t)
+    {
+        warn(msg, t);
+    }
+
+    public static Level intToLevel(int level)
+    {
+        if (level >= JettyLogger.OFF)
+            return Level.ERROR;
+        if (level >= Level.ERROR.toInt())
+            return Level.ERROR;
+        if (level >= Level.WARN.toInt())
+            return Level.WARN;
+        if (level >= Level.INFO.toInt())
+            return Level.INFO;
+        if (level >= Level.DEBUG.toInt())
+            return Level.DEBUG;
+        if (level >= Level.TRACE.toInt())
+            return Level.TRACE;
+        return Level.TRACE; // everything else
+    }
+
+    public static String levelToString(int level)
+    {
+        if (level >= JettyLogger.OFF)
+            return "OFF";
+        if (level >= Level.ERROR.toInt())
+            return "ERROR";
+        if (level >= Level.WARN.toInt())
+            return "WARN";
+        if (level >= Level.INFO.toInt())
+            return "INFO";
+        if (level >= Level.DEBUG.toInt())
+            return "DEBUG";
+        if (level >= Level.TRACE.toInt())
+            return "TRACE";
+        return "OFF"; // everything else
+    }
+
+    @SuppressWarnings("StringBufferReplaceableByString")
+    @Override
+    public String toString()
+    {
+        final StringBuilder sb = new StringBuilder();
+        sb.append(JettyLogger.class.getSimpleName());
+        sb.append(':').append(name);
+        sb.append(":LEVEL=").append(levelToString(level));
+        return sb.toString();
+    }
+
+    private JettyLoggingEvent asEvent(Level level, String msg)
```

Comment:
See above, all these `asEvent()` are not necessary.

Reply:
JettyLoggingEvent has been removed.